src/hypertable_cache.c

```diff
@@ -137,6 +138,34 @@ ts_hypertable_cache_get_entry(Cache *cache, Oid relid)
 	return ts_hypertable_cache_get_entry_with_table(cache, relid, NULL, NULL);
 }
 
+/*
+ * Returns hypertable by checking the given cache first.
+ * If hypertable is not found, fails with an error.
+ */
+Hypertable *
+ts_hypertable_cache_get_entry_always(const Oid relid, Cache *const cache)
+{
+	Hypertable *const ht= ts_hypertable_cache_get_entry(cache, relid);
+
+	if (ht == NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_TS_HYPERTABLE_NOT_EXIST),
```

Comment:
Strictly speaking, there is a difference between "table does not exist" and "table is wrong type". The two error conditions belong to different error categories and I think we should be careful and make sure to not mix them up.

Reply:
I am not sure what you are out for. I believe the error code was created exactly for this case that hypertable is not found for the given OID. Which means that the database object is not a hypertable. I believe this discussion is outside of this PR scope, since it doesn't introduce new error code, but uses existing one..

Reply:
I agree with @mkindahl that we might want to distinguish between these cases:

* The relid is invalid (InvalidOid)  (ERRCODE_UNDEFINED_OBJECT)
* The relid is valid but is not a table (ERRCODE_WRONG_OBJECT_TYPE)
* The relid is a valid table relation, but is not a hypertable (TS_HYPERTABLE_DOES_NOT_EXIST), 
* The relid is a valid hypertable (no error)

We might choose to not distinguish between the first two errors, however. But since this is now all in one place, it is easy to be specific.

Reply:
Some cases returned `ERRCODE_INVALID_PARAMETER_VALUE` and some returned `ERRCODE_TS_HYPERTABLE_NOT_EXIST`. Now they all return `ERRCODE_TS_HYPERTABLE_NOT_EXIST`.

Reply:
@erimatnor Thank you for clarifying. I see the point. Distinguishing between option 1 and 2 is easy as they are separate code paths.

Reply:
> Some cases returned `ERRCODE_INVALID_PARAMETER_VALUE` and some returned `ERRCODE_TS_HYPERTABLE_NOT_EXIST`. Now they all return `ERRCODE_TS_HYPERTABLE_NOT_EXIST`.

Looking to the actual cases it seems that the choice between the error codes was arbitrary and subjective. However, I agree with Erik's point.

Reply:
Note that I am not sure we'd like to use `ERRCODE_INVALID_PARAMETER_VALUE` here since we don't have the context in this function to know whether the call came due to a user-invoked function. For instance, it would be a bit weird to raise this error on an insert or non-function operation.

Reply:
Implemented suggestion from @erimatnor

Reply:
> > Some cases returned `ERRCODE_INVALID_PARAMETER_VALUE` and some returned `ERRCODE_TS_HYPERTABLE_NOT_EXIST`. Now they all return `ERRCODE_TS_HYPERTABLE_NOT_EXIST`.
> 
> Looking to the actual cases it seems that the choice between the error codes was arbitrary and subjective. However, I agree with Erik's point.

Maybe, but the key point is that there are different reasons why the hypertable could not be found, which is what I tried to highlight. @erimatnor did a better job here. :)